uberenv.py

```diff
@@ -465,7 +468,7 @@ def install(self):
         if self.opts["ignore_ssl_errors"]:
             install_cmd += "-k "
         install_cmd += "dev-build -d {} ".format(self.pkg_src_dir)
-        if not self.opts["install"]:
+        if not self.opts["install"] and self.pkg_final_phase:
```

Comment:
At this point, shouldn’t we inform the user that "not install" mode expects a final phase? Or document the behavior?

Uberenv will continue and actually build the project in spack using the current source. The behavior is thus slightly different from what is used to be: build a fake project.

Reply:
I am not sure what the correct answer is here. If you don't have a final phase and call uberenv (without --install) do we want it to fail?

Right now everyone I know has a final phase but I am not sure that's going to be the case forever.  I was just working around splitting the devtools for my two projects out and this seemed the best thing to do since the devtools bundlepackage doesn't have any phases.

Thoughts @cyrush ?

Reply:
for now I would opt for an error if --install is not selected and there is no final phase.

It's not clear what we are asking uberenv to do in that case.

Reply:
I am fine with whatever.  I hide all this in scripts.  I was just wondering what we feel is the best workflow for users.

To be clear, if we have a project/package foo that does not have a final phase.  We want it to error out unless you say --install.  Instead of defaulting to just building everything?

Is this what you want?

uberenv.py foo
 Error: No final phase declared for package 'foo'.  Either declare what phase to stop after on the command line or in the project.json.  Alternately, the command line option "--install" will build your complete package.

Reply:
My 2 cents...

@white238 I understand you concern. It does not surprise me much however since the default behavior of uberenv _is not_ --install. To me, this choice means that Uberenv is primarily meant to deal with dependency installs.

I’ll let the creator answer this question though.